### PR TITLE
EmailCommunication-Scripts-Enhancement

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: demisto/etc/issues/28837
related: https://github.com/demisto/etc/issues/29245

### SendEmailReply:
SendEmailReply to execute `reply-mail` instead of `send-mail`

### PreprocessEmail
Added a list of the email attachments to the `set_email_reply` return value.
Replaced the source of the email attachments with entry ids in order to view inline images.
Added the `update_latest_message_field` function which set the email related incident context with the latest email reply id. 
Updated the `create_email_html` function to also delete the email original message quote from the html body.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py [hunk 1]
```diff
@@ -108,36 +162,68 @@ def get_attachments_using_instance(email_related_incident, labels):
...
+def get_incident_related_files(incident_id):
+    """Get the email reply attachments after they were uploaded to the server and saved
+    to context of the email reply related incident.
+
+    Args:
+        incident_id (str): The ID of the incident whose context we'd like to get.
+    """
+    try:
+        res = demisto.executeCommand("getContext", {'id': incident_id})
+        return res[0]['Contents'].get('context', {}).get('File', [])
```
Comment (yaakovi) [inline Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py:174]: use `dict_safe_get`

File: Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py [hunk 2]
```diff
@@ -108,36 +162,68 @@ def get_attachments_using_instance(email_related_incident, labels):
         demisto.debug('Attachments could only be retrieved from EWS v2 or Gmail')
 
 
+def get_incident_related_files(incident_id):
+    """Get the email reply attachments after they were uploaded to the server and saved
+    to context of the email reply related incident.
+
+    Args:
+        incident_id (str): The ID of the incident whose context we'd like to get.
```
Comment (yaakovi) [inline Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py:170]: why `incident_id` here is a string and in the `update_latest_message_field` it is a number

File: Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py [hunk 3]
```diff
@@ -1,43 +1,97 @@
...
+    """
+
+    # Removing the conversation's history
+    for marker in QUOTE_MARKERS:
+        index = email_html.find(marker)
+        if index != -1:
+            email_html = f'{email_html[:index]}</body></html>'
+
+    # Replacing the images' sources from
+    for entry_id in entry_id_list:
```
Comment (yaakovi) [inline Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py:27]: how does entry_id look like? what is the first object? what is the second?
split it into different variables: `for image_name, image_path in entry_id_list:` (I guessed the meanings)

File: Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py [hunk 4]
```diff
@@ -1,43 +1,97 @@
...
 import json
+import re
 
+ERROR_TEMPLATE = 'ERROR: PreprocessEmail - {function_name}: {reason}'
+QUOTE_MARKERS = ['<div class="gmail_quote">',
+                 '<hr tabindex="-1" style="display:inline-block; width:98%"><div id="divRplyFwdMsg"']
 
-def add_entries(single_reply, email_related_incident):
+
+def create_email_html(email_html='', entry_id_list=[]):
```
Comment (yaakovi) [inline Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py:11]: don't use mutable objects as a default value, this is a bug.
use None and handle that case inside the function.
https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

File: Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail_test.py [hunk 5]
```diff
@@ -14,5 +24,198 @@ def test_set_email_reply():
...
+        - Return False to drop the newly created incident and attach the relevant data to the existing
+        email related incident.
+    """
+    import PreprocessEmail
+    from PreprocessEmail import main
+    incident = util_load_json('test_data/get_incident_details_result.json')
+    mocker.patch.object(demisto, 'incident', return_value=incident)
+    mocker.patch.object(PreprocessEmail, 'get_incident_by_query',
+                        return_value=util_load_json('test_data/email_related_incident_response.json'))
+    # mocker.patch.object(PreprocessEmail, 'check_incident_status')
```
Comment (yaakovi) [inline Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail_test.py:216]: remove if not needed

**Review by yaakovi — CHANGES_REQUESTED** [review]

**Comment by yaakovi** [comment]
please schedule a demo after fixing the comments

**Comment by kirbles19** [comment]
@Bargenish Done.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**